node_exporter.go

```diff
@@ -97,6 +99,9 @@ func main() {
 		log.Infof(" - %s", n)
 	}
 
+	// Use our shared code to run server and exit on error. Upstream's code below will not be executed.
+	exporter_shared.RunServer("Node", *listenAddress, *metricsPath, handler)
```

Comment:
That line calls the`handler` function defined at https://github.com/percona/node_exporter/pull/10/files#diff-96274f342a5b55fb60671bb66a2ea228R36. That completely skips basic authentication.